queue/rabbitmq.go

```diff
@@ -1,22 +1,25 @@
 package queue
 
 import (
+	"encoding/json"
 	"errors"
 	"fmt"
+
 	"github.com/loafoe/go-rabbitmq"
 	"github.com/philips-software/go-hsdp-api/logging"
 	"github.com/streadway/amqp"
 )
 
 var (
-	Exchange   = "logproxy"
-	RoutingKey = "new.rfc5424"
-	ErrInvalidProducer = errors.New("RabbitMQ producer is nil or invalid")
+	Exchange             = "logproxy"
+	RoutingKey           = "new.rfc5424"
+	DeadLetterRoutingKey = "deadletter.resource"
```

Comment:
For what i understand there is no DL-Xchange present (https://www.rabbitmq.com/dlx.html) and that might hamper accurate execution. 

Typically Rabbit dead lettering uses a separate Exchange with a header to cause expiry and re-queue automatically. 
Also the declared queue needs to have a max-length limit on it and not be ephemeral. 

it should also look at `x-death` to limit number of retries. 

envisioned flow:
(MSG) -> (Ex:logproxy) -> (Q: RFC5424QueueName()) -Error-> (Ex: DLX.logproxy) -> (Q: DLX.RFC5424QueueName()) -MSG.expiry-> (Ex:Logproxy)

Since any error causes the message to be re-queued, the DeadLetter handler needs to inspect the `X-Death` header and if it is more than a limit send it to a final Queue for forensic purposes. Note the final queue needs to be declared with Max-Length constraint to prevent resource exhaustion.

Reply:
After the first pass the message is no longer in RFC5424 but a logging.Resource, so a different worker needs to pick up the retry handling. I've already add a ResourceWorker. Now need to figure out if Logproxy can set up the complete DLX setup (exchange, policies, etc). The flow would be:

(RFC5424MSG) -> (Ex:logproxy) -> (Q: RFC5424QueueName()) -> (W:RFC5424Worker) -Error-> (Ex: DLX.logproxy) -> (Q: DLX.ResourceQueueName()) -> (W:ResourceWorker) -Error-> (Ex: DLX.logproxy) -> (Q: DLX.ResourceQueueName()) -MSG.expiry

Looks like clients can define DLX just fine.